handler-ssl-ocsp/src/main/java/io/netty/handler/ssl/ocsp/OcspValidationEvent.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl.ocsp;
+
+import org.bouncycastle.cert.ocsp.BasicOCSPResp;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+public final class OcspValidationEvent {
+
+    private final BasicOCSPResp response;
+
+    public OcspValidationEvent(BasicOCSPResp response) {
+        this.response = checkNotNull(response, "BasicOCSPResp");
+    }
+
+    public BasicOCSPResp response() {
+        return response;
+    }
+
+    /**
+     * Returns {@link Boolean#TRUE} if server certificate is valid else {@link Boolean#FALSE}
+     */
+    public boolean isCertificateValid() {
+        return response.getResponses()[0].getCertStatus() == null;
```

Comment:
Since we request only 1 certificate status at a time, we should only receive 1 response. We validate the number of responses too, i.e. if it's not 1 then it will throw an exception.

```java
            int responses = basicResponse.getResponses().length;
            if (responses != 1) {
                throw new IllegalArgumentException("Expected number of responses was 1 but got: " + responses);
            }
```

Reply:
So its safe to access the array directly without any more checks.